src/components/CartPage/CartItem/CartItem.js

```diff
@@ -7,94 +7,165 @@ import './CartItem.scss';
 
 const CN = 'cart-item';
 
-const CartItem = (props) => {
-  const {
-    item, changeQuantity, userCart, removeItemFromCart, className
-  } = props;
-  const {
-    _id, title, images, price, quantity
-  } = item;
-  const userCartItem = userCart.find((cartItem) => cartItem.id === _id) || {};
-  const { chosenQuantity, color, size } = userCartItem;
+class CartItem extends Component {
+  constructor(props) {
+    super(props);
 
-  const inc = 1;
-  const dec = -1;
+    this.state = {
+      inc: 1,
+      dec: -1,
+      convertedPrice: 0
+    };
 
-  let userCartNew = [...userCart];
+    this.changeAmount = this.changeAmount.bind(this);
+    this.onRemoveFromCart = this.onRemoveFromCart.bind(this);
+    this.handleQuantity = this.handleQuantity.bind(this);
+    this.renderProductDescription = this.renderProductDescription.bind(this);
+    this.renderCounter = this.renderCounter.bind(this);
+  }
+
+  async componentDidMount() {
+    this.getConvertedPrice();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { currencyKey } = this.props;
+    if (currencyKey !== prevProps.currencyKey) {
+      this.getConvertedPrice();
+    }
+  }
+
+  onRemoveFromCart() {
+    const { userCart, removeItemFromCart, item } = this.props;
+    const { _id } = item;
+    let userCartNew = [...userCart];
+
+    userCartNew = userCart.filter((product) => product.id !== _id);
+    removeItemFromCart(userCartNew);
+  }
+
+  getConvertedPrice() {
+    const { item, currencyKey } = this.props;
+    const { price } = item;
+    const currencyPrice = (price * currencyKey).toFixed(0);
```

Comment:
this can be calculated inside this.setState. No need to have seperate variable for this

Reply:
I dont think that it's good idea to calculate inside of this.setState. Why do we always have to put all calculations into constants and use these constants?

`const currencyPrice = (price * currencyKey).toFixed(0);
    this.setState({
      convertedPrice: currencyPrice
   });`

Reply:
Well yes, but in this case, it's clear enough what you are doing - so you could shorten it a bit